sys/fmt/fmt.c

```diff
@@ -243,6 +243,51 @@ size_t fmt_s16_dfp(char *out, int16_t val, unsigned fp_digits)
     return pos;
 }
 
+size_t fmt_s32_dfp(char *out, int32_t val, unsigned fp_digits)
+{
+    int32_t absolute, divider;
+    size_t pos = 0;
+    size_t div_len, len;
+    unsigned e;
+    char tmp[9];
+
+    if (fp_digits > 9) {
+        return 0;
+    }
+    if (fp_digits == 0) {
+        return fmt_s32_dec(out, val);
+    }
+    if (val < 0) {
+        if (out) {
+            out[pos++] = '-';
+        }
+        val *= -1;
+    }
+
+    e = pwr(10, fp_digits);
```

Comment:
that's what I meant with 

> could adapt to #7263 and #7366 

applies for both your comments 😉 but yes at least for the latter, I think its better to use ten map